File: 007.跨端APP应用（移动端开发工程师）/uniapp-project/utils/nav-intent.ts
/**
 * 跨 tabBar 跳转时携带一次性意图(uni.switchTab 不支持 url query)。
 * 跳转前调用 setPendingMonth(),目标页 onLoad 里 consumePendingMonth() 读取后清空。
 */
let pendingMonth: string | null = null

export function setPendingMonth(m: string): void {
  pendingMonth = m
}

export function consumePendingMonth(): string | null {
  const m = pendingMonth
  pendingMonth = null
  return m
}

File: 007.跨端APP应用（移动端开发工程师）/flutter-project/lib/core/router/nav_intent.dart
/// 对齐 utils/nav-intent.ts — go_router 不支持向 tabBar 路由传 query,
/// 改用模块级 pendingMonth 单例,首页"查看全部"调用 setPendingMonth 后
/// switchTab,流水页在 onShow 时调 consumePendingMonth 取回并清空。
String? _pendingMonth;
void setPendingMonth(String m) => _pendingMonth = m;
String? consumePendingMonth() {
  final m = _pendingMonth;
  _pendingMonth = null;
  return m;
}
